msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -0,0 +1,49 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Xamarin.iOS.Tasks {
+	[TestFixture]
+	public class LocalizationStringTest : TestBase {
+		[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
+		[TestCase ("de-DE", "Unbekanntes Imageformat.")]
+		[TestCase ("en-US", "Unknown image format.")]
+		[TestCase ("es-ES", "formato de imagen desconocido.")]
+		[TestCase ("fr-FR", "format d'image inconnu.")]
+		[TestCase ("it-IT", "Formato immagine sconosciuto.")]
+		[TestCase ("ja-JP", "の読み込みでエラーが発生しました: 画像の形式が不明です。")]
+		[TestCase ("ko-KR", "을(를) 로드하는 동안 오류 발생: 알 수 없는 이미지 형식입니다.")]
+		[TestCase ("pl-PL", "nieznany format obrazu.")]
+		[TestCase ("pt-BR", "formato de imagem desconhecido.")]
+		[TestCase ("ru-RU", "неизвестный формат изображения.")]
+		[TestCase ("tr-TR", "yüklenirken hata oluştu: Görüntü biçimi bilinmiyor.")]
+		[TestCase ("zh-CN", "时出错: 未知图像格式")]
+		[TestCase ("zh-TW", "時發生錯誤: 未知的映像格式。")]
+		public void AllSupportedTranslations (string culture, string errorMessage)
+		{
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+			CultureInfo newCulture;
+			try {
+				newCulture = new CultureInfo (culture);
+				Thread.CurrentThread.CurrentUICulture = newCulture;
+			}
+			catch (System.Globalization.CultureNotFoundException) {
+				Assert.IsTrue (false, culture + ": is not a valid culture. ");
+			}
+
+			var task = CreateTask<CollectITunesArtwork> ();
+			task.ITunesArtwork = new TaskItem [] { new TaskItem (Assembly.GetExecutingAssembly ().Location) };
+
+			Assert.IsFalse (task.Execute (), "Execute failure");
+			Assert.AreEqual (1, Engine.Logger.ErrorEvents.Count, "ErrorCount");
+			bool isTranslated = Engine.Logger.ErrorEvents[0].Message.Contains (errorMessage);
+			Assert.IsTrue (isTranslated, culture + ": is not supported correctly. ");
+
+			Thread.CurrentThread.CurrentUICulture = originalCulture;
```

Comment:
This should be in a finally clause, so that it gets restored even if the test fails:

```csharp
var originalCulture = Thread.CurrentThread.CurrentUICulture;
try {
    var newCulture = new CultureInfo (culture);
    ...
} finally {
    Thread.CurrentThread.CurrentUICulture = originalCulture;
}

Reply:
So with a try, catch, finally statement, if the code goes to the catch part where it's hitting my assert and failing, doesn't the finally not run?

Reply:
So because of this ^, I just made it a try catch and restored the original culture in both of the situations. I don't think it very pretty though so I'm open to suggestions!

Reply:
The finally clause *always* runs, even if there's a catch clause (and that catch clause was executed because there was an exception).

Reply:
Ah okay, so I may have been misunderstanding the original problem, my apologies. So if the CultureNotFoundException occurs, we do not need to reset the original culture? If so, why not? Or even if there is an exception in the catch clause, the finally clause is still ran?

Reply:
The finally clause is ran even if there's an exception in the catch clause.